### ISSUE TITLE
Unable to reference to envoy grpc proto in custom filter when using envoy 1.8.0

### ISSUE THREAD

**Issue by aejeet** [envoyproxy/envoy#5334]
**_Title:_** Unable to reference to envoy grpc proto in custom filter when using envoy 1.8.0

**_Description:_**
I am using the http-filter-example template to implement a custom envoy http filter of my own. In the filter config i need a reference to api/envoy/api/v2/core/grpc_service.proto. **The following BUILD script was working fine with envoy 1.7.1. But it fails to compile with envoy 1.8.0.** 

I am new to bazel, and any help will be highly appreciated. I need 1.8.0 for some of the additional capabilities and staying on 1.7.1 is not an option

**_Additional Details:_** 
**My filter config file filter_config.proto**
```
syntax = "proto3";
package myproto;

import "validate/validate.proto";
import "envoy/api/v2/core/grpc_service.proto";

message EpmpCmds {
  // The external  gRPC service configuration 
  envoy.api.v2.core.GrpcService cnc_service = 1;

 // additional config -  mostly boolean fields
}
```

**My BUILD file**
```
package(default_visibility = ["//visibility:public"])

load(
    "@envoy//bazel:envoy_build_system.bzl",
    "envoy_cc_binary",
    "envoy_cc_library",
    "envoy_cc_test_library",
    "envoy_cc_test",
)

load("@envoy_api//bazel:api_build_system.bzl", "api_proto_library")

api_proto_library(
    name = "filter_config_proto",
    srcs = ["filter_config.proto"],
    deps = [
        "@envoy_api//envoy/api/v2/core:grpc_service",
    ],
)

// additional lines
```

**Encountered ERROR**
```
DEBUG: /private/var/tmp/_bazel_ajit_verma/1f2dd7d102aa45a25d81491054a38483/external/envoy/bazel/repositories.bzl:120:5: External dep build exited with return code: 0
ERROR: /Users/ajit_verma/Workspace/custom-envoy/epmp/filters/epmp_cmds/BUILD:13:1: target '@envoy_api//envoy/api/v2/core:grpc_service' is not visible from target '//epmp/filters/epmp_cmds:config_proto'. Check the visibility declaration of the former target if you think the dependency is legitimate
ERROR: Analysis of target '//:envoy' failed; build aborted: Analysis of target '//epmp/filters/epmp_cmds:config_proto' failed; build aborted
```

**Comment by alyssawilk** [comment]
I think the raw libraries are generally default-private (as part of the fix to #3756) but can you try "@envoy_api//envoy/api/v2/core:grpc_service_cc" and see if that fixes it for you?

**Comment by aejeet** [comment]
> I think the raw libraries are generally default-private (as part of the fix to #3756) but can you try "@envoy_api//envoy/api/v2/core:grpc_service_cc" and see if that fixes it for you?

I think i had tried this earlier. Tried again but no luck. Here is the new error
```
DEBUG: /private/var/tmp/_bazel_ajit_verma/1f2dd7d102aa45a25d81491054a38483/external/envoy/bazel/repositories.bzl:120:5: External dep build exited with return code: 0
ERROR: /Users/ajit_verma/Workspace/custom-envoy/epmp/filters/epmp_cmds/BUILD:13:1: in deps attribute of proto_library rule //epmp/filters/epmp_cmds:config_proto: cc_library rule '@envoy_api//envoy/api/v2/core:grpc_service_cc' is misplaced here (expected proto_library). Since this rule was created by the macro 'api_proto_library', the error might have been caused by the macro implementation in /private/var/tmp/_bazel_ajit_verma/1f2dd7d102aa45a25d81491054a38483/external/envoy_api/bazel/api_build_system.bzl:104:16
ERROR: /Users/ajit_verma/Workspace/custom-envoy/epmp/filters/epmp_cmds/BUILD:13:1: no such target '@envoy_api//envoy/api/v2/core:grpc_service_cc_cc': target 'grpc_service_cc_cc' not declared in package 'envoy/api/v2/core' (did you mean 'grpc_service_cc'?) defined by /private/var/tmp/_bazel_ajit_verma/1f2dd7d102aa45a25d81491054a38483/external/envoy_api/envoy/api/v2/core/BUILD and referenced by '//epmp/filters/epmp_cmds:config_proto_cc'
ERROR: Analysis of target '//:envoy' failed; build aborted: Analysis of target '//epmp/filters/epmp_cmds:config_proto' failed; build aborted
INFO: Elapsed time: 1.277s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (28 packages loaded)
```

**It is easily reproducible using following simple steps**
1. Clone https://github.com/envoyproxy/envoy-filter-example
2. Go to base folder and `git checkout 4d135eb2e4b30cc73b0d1a87ac046b68d3b0c01d`  (checkout commit that updated the envoy submodule to v1.8.0)
3. Modify `http-filter-example/http_filter.proto` to include `import "envoy/api/v2/core/grpc_service.proto"` and `envoy.api.v2.core.GrpcService grpc_service = 5;`
4. Modify `http-filter-example/BUILD` to include
```
 api_proto_library(
     name = "http_filter_proto",
-    srcs = ["http_filter.proto"]
+    srcs = ["http_filter.proto"],
+    deps = [
+        "@envoy_api//envoy/api/v2/core:grpc_service_cc",
+    ],
 )

```
5. Run `git submodule update --init`
6. Run `bazel build //http-filter-example:envoy`

**Comment by aejeet** [comment]
@lizan Using `_export` suffix seem to fix the visibility issue.  Thanks for the quick help.
